draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -1470,7 +1472,10 @@ user agent MUST process the cookie as follows:
 
     1.  Let the domain-attribute be the empty string.
 
-8.  If the user agent is configured to reject "public suffixes" and the
+8.  If the domain-attribute contains a character that is not in the range of {{USASCII}}
+    characters (%x00 - %x7F), abort these steps and ignore the cookie entirely.
```

Comment:
It feels odd to include characters we specifically avoid elsewhere as valid (such as %x00). I guess the earlier parts of the spec handle those cases where necessary so this should be ok, it's strikes me as add.

Just noting this for posterity. No change required.

Reply:
I think that we can use a narrower range for acceptance here that will ensure that rejected characters (%x00) don't get included.  A domain name can be just limited to LDH (letter, digit, hyphen) + '.'.

It seems we're very deliberately avoiding mention of IP addresses, so we can probably not specify tighter validation rules (along the lines of what RFC 1034 specifies), but somewhat tighter is probably wise.

Reply:
I agree it's a bit weird to explicitly list those. I'm not sure to what extent it's better to just remove the explicit mention of the character range vs. having a narrower definition. The UA will perform an equality check to a canonicalized URL host right after this step, so there's not a lot of worth to UAs in checking for LDH explicitly. Browsers have very optimized isASCII functions that would have to be sidestepped for this.

(Even this step is mostly added to make it clear to UAs that they should not perform canonicalization on the domain attribute, strictly speaking it wouldn't be necessary since the spec doesn't mandate canonical/A-label domain attributes)

If we do go for referencing LDH it might be useful to declare it in the terminology, I filed #1979 in any case.

Reply:
I like this simpler check for that reason too. We don't actually want to reject `..` or some such at this point.

(IP addresses we should tackle as part of #1939.)

Reply:
I'm OK with that, but this is a good point to point out that the domain canonicalization will ensure that invalid domain names will be rejected.  The oddness (and inconsistency) is pretty apparent and you've had two people tripped up by it already.

Reply:
Added a note to help clear the confusion, lmk what you think :)